src/network/mod.rs

```diff
@@ -47,7 +47,8 @@ impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match *self {
             Error::Io(ref e) => fmt::Display::fmt(e, f),
-            Error::SocketMutexPoisoned | Error::SocketNotConnectedToPeer => f.write_str(error::Error::description(self)),
+            Error::SocketMutexPoisoned => f.write_str("socket mutex was poisoned"),
+            Error::SocketNotConnectedToPeer => f.write_str("socket mutex was poisoned"),
```

Comment:
this message doesn't seem right

Reply:
You're right! 
Thanks for catching this, I re-reviewed my PR to make sure I didn't miss anything else